tools/ci_connector_ops/ci_connector_ops/qa_checks.py

```diff
@@ -2,65 +2,106 @@
     """Check if a markdown file with connector documentation is available 
-    in docs/integrations/<connector-type>/<connector-name>.md
+    in docs/integrations/<connector-type>s/<connector-name>.md
 
     Args:
-        connector_name (str): The connector name
+        connector (Connector): a Connector dataclass instance.
 
     Returns:
         bool: Wether a documentation file was found.
     """
-    # TODO implement
-    return True
 
-def check_changelog_entry_is_updated(connector_name: str) -> bool:
+    return connector.documentation_file_path.exists()
+
+def check_documentation_follows_guidelines(connector: Connector) -> bool:
+    """Documentation guidelines are defined here https://hackmd.io/Bz75cgATSbm7DjrAqgl4rw"""
+    follows_guidelines = True
+    with open(connector.documentation_file_path) as f:
+        doc_lines = [l.lower() for l in f.read().splitlines()]
+    if not doc_lines[0].startswith("# "):
+        print("The connector name is not used as the main header in the documentation.")
+        follows_guidelines = False
+    if connector.definition: # We usually don't have a definition if the connector is not published.
+        if doc_lines[0].strip() != f"# {connector.definition['name'].lower()}":
+            print("The connector name is not used as the main header in the documentation.")
+            follows_guidelines = False
+    elif not doc_lines[0].startswith("# "):
+        print("The connector name is not used as the main header in the documentation.")
+        follows_guidelines = False
+
+    expected_sections = [
+        "## Prerequisites", 
+        "## Setup guide",
+        "## Supported sync modes",
+        "## Supported streams",
+        "## Changelog"
+    ]
+
+    for expected_section in expected_sections:
+        if expected_section.lower() not in doc_lines:
+            print(f"Connector documentation is missing a '{expected_section.replace('#', '').strip()}' section.")
+            follows_guidelines = False
+    return follows_guidelines
+
+def check_changelog_entry_is_updated(connector: Connector) -> bool:
     """Check that the changelog entry is updated for the latest connector version
     in docs/integrations/<connector-type>/<connector-name>.md
 
     Args:
-        connector_name (str): The connector name
+        connector (Connector): a Connector dataclass instance.
 
     Returns:
         bool: Wether a the changelog is up to date.
     """
-    # TODO implement
-    return True
-
-def check_connector_icon_is_available(connector_name: str) -> bool:
+    if not check_documentation_file_exists(connector):
+        return False
+    with open(connector.documentation_file_path) as f:
+        after_changelog = False
+        for line in f:
+            if "# changelog" in line.lower():
+                after_changelog = True
+            if after_changelog and connector.version in line:
+                return True
+    return False
```

Comment:
nit: This check is great to have, though having it run as the first part of the /check command feels like it might be a _bit_ of a DX hit. I think in general people tend to work on changes and deal with bumping versions and changelog later on in the process, but from looking at the example runs, this is running as part of the integration-tests step and blocks other SATs from running.

This isn't really a comment on this specific set of changes, but I think it would be nice if we could separate the integration-tests step from these QA checks (and potentially have them run in parallel), so people can still /test and see the SAT results. In my mind integration tests and these QA checks are somewhat separate.